Formula/l/lune.rb

```diff
@@ -15,10 +15,11 @@ class Lune < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "53e9d5eadb78c68c2fd20d7184609e1a560f8efe3ed6e8b2f144194e8cf3fe9b"
   end
 
+  depends_on "cmake" => :build
```

Comment:
Why do we need CMake?

Reply:
Cmake is a dependency within its rust crates: https://github.com/lune-org/lune/blob/7e784ba361ba568f76aa17d0d544d170db86dfc0/Cargo.lock#L502

I just checked what happens if it's omitted, and it does throw this error:
```
  failed to execute command: No such file or directory (os error 2)
  is `cmake` not installed?

  build script failed, must exit now
```

Reply:
Yes, but I'm more interested in what it's building and whether it's duplicating a formula we already have.

Reply:
```
error: failed to run custom build command for `libz-ng-sys v1.1.15`

Caused by:
  process didn't exit successfully: `/private/tmp/lune-20240512-89103-nq6fiz/lune-0.8.4/target/release/build/libz-ng-sys-f6ac78e7be068455/build-script-build_zng` (exit status: 101)
  --- stdout
  CMAKE_TOOLCHAIN_FILE_aarch64-apple-darwin = None
  CMAKE_TOOLCHAIN_FILE_aarch64_apple_darwin = None
  HOST_CMAKE_TOOLCHAIN_FILE = None
  CMAKE_TOOLCHAIN_FILE = None
  CMAKE_GENERATOR_aarch64-apple-darwin = None
  CMAKE_GENERATOR_aarch64_apple_darwin = None
  HOST_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_aarch64-apple-darwin = None
  CMAKE_PREFIX_PATH_aarch64_apple_darwin = None
  HOST_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = Some("/opt/homebrew")
  CMAKE_aarch64-apple-darwin = None
  CMAKE_aarch64_apple_darwin = None
  HOST_CMAKE = None
  CMAKE = None
  running: cd "/private/tmp/lune-20240512-89103-nq6fiz/lune-0.8.4/target/release/build/libz-ng-sys-0387c71c4fe444c1/out/build" && CMAKE_PREFIX_PATH="/opt/homebrew" "cmake" "/Users/rui/Library/Caches/Homebrew/cargo_cache/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.15/src/zlib-ng" "-DCMAKE_OSX_ARCHITECTURES=arm64" "-DBUILD_SHARED_LIBS=OFF" "-DZLIB_COMPAT=OFF" "-DZLIB_ENABLE_TESTS=OFF" "-DWITH_GZFILEOP=ON" "-DCMAKE_INSTALL_PREFIX=/private/tmp/lune-20240512-89103-nq6fiz/lune-0.8.4/target/release/build/libz-ng-sys-0387c71c4fe444c1/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-darwin -mmacosx-version-min=14.4" "-DCMAKE_C_COMPILER=/opt/homebrew/Library/Homebrew/shims/mac/super/clang" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-darwin -mmacosx-version-min=14.4" "-DCMAKE_CXX_COMPILER=/opt/homebrew/Library/Homebrew/shims/mac/super/clang++" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-darwin -mmacosx-version-min=14.4" "-DCMAKE_ASM_COMPILER=/opt/homebrew/Library/Homebrew/shims/mac/super/clang" "-DCMAKE_BUILD_TYPE=MinSizeRel"

  --- stderr
  thread 'main' panicked at /Users/rui/Library/Caches/Homebrew/cargo_cache/registry/src/index.crates.io-6f17d22bba15001f/cmake-0.1.50/src/lib.rs:1098:5:

  failed to execute command: No such file or directory (os error 2)
  is `cmake` not installed?

  build script failed, must exit now
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```

Reply:
same deal as:
- https://github.com/Homebrew/homebrew-core/pull/155864
- upstream feature request, https://github.com/rust-lang/libz-sys/issues/147